### DISCUSSION TITLE
GraphQL query for future events does not work

### DISCUSSION THREAD

**Discussion by denisyilmaz** [solspace/craft-calendar#42] [Questions & Answers] [answered]
First of all I'm very happy that you introduced the new GraphQL feature! comes at the perfect timing for an upcoming launch.

Unfortunatelty I am unable to query for future events only.

**Steps to reproduce**

1. Query by  entries shows all entries starting the beginning of the calendar
```gql
{
  solspace_calendar {
    version
    calendar {
      id
      events {
        id
        startDate
      }
    }
  }
}
```

results in a list of entries:

```json
{
  "data": {
    "solspace_calendar": {
      "version": "3.3.0",
      "calendar": {
        "id": 1,
        "events": [
          {
            "id": 64,
            "startDate": "2020-05-03T00:00:00+00:00"
          },
          {
            "id": 64,
            "startDate": "2020-06-07T00:00:00+00:00"
          },
          {
            "id": 64,
            "startDate": "2020-07-05T00:00:00+00:00"
          },
          {
            "id": 64,
            "startDate": "2020-08-02T00:00:00+00:00"
          }
        ]
      }
    }
  }
}

```

2. But querying with `events(startsAfter:"now")` results in an empty array:

```gql
{
  solspace_calendar {
    version
    calendar {
      id
      events(startsAfter:"now") {
        id
        startDate
      }
    }
  }
}
```

Result:

```json
{
  "data": {
    "solspace_calendar": {
      "version": "3.3.0",
      "calendar": {
        "id": 1,
        "events": []
      }
    }
  }
}
```
 
I also tested with `events(startsAfter:"2021-05-28T00:00:00+00:00")` and `events(startsAfter:"2021-05-28")` with same result.

**Expected behavior**

Extected result would be a list of events starting after today.

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 3.6.15
- Calendar Version: 3.3.0
- Fresh Install or Upgrade: Upgrade

**Additional context**

- Running locally with Nitro 2.0.8

**Comment by gustavs-gutmanis** [comment, marked as answer]
Glad to hear @denisyilmaz 

If you want to query events based on their dates, there are filtering functions available.
I just noticed that the docs don't have the GQL examples released yet, I'll post them here once they are.

A small example how to filter in your case:
```gql
{
  solspace_calendar {
    events (rangeStart: "now") {
      id
      title
      startDate
    }
  }
}
```

The `startsAfter` filter is for filtering base events, not the occurrences.

**Reply by kjmartens** [reply]
The new documentation for GraphQL support in Calendar 3.3+ is now available: https://docs.solspace.com/craft/calendar/v3/developer/graphql.html
